src/pfe/file-watcher/server/test/functional-test/suites/project/tests/update-status.ts

```diff
@@ -0,0 +1,206 @@
+                        let itData = _.cloneDeep(testData);
+                        itData = makeStatusUpdateObj(itData, statusTypes[statusType]["parameters"], values);
+
+                        const result = _.cloneDeep(statusTypes[statusType]["result"]);
+                        result[eventKeys[1]] = state;
+
+                        const info: any = await updateStatus(itData);
+                        expect(info);
+                        expect(info.statusCode).to.equal(200);
+
+                        if (eventKeys) {
+                            const targetEvent = eventConfigs.events.statusChanged;
+                            let eventFound = false;
+                            let event: any;
+                            await new Promise((resolve) => {
+                                const timer = setInterval(() => {
+                                    const events = socket.getAllEvents();
+                                    if (events && events.length >= 1) {
+                                        event =  events.filter((value) => {
+                                            if (value.eventName === targetEvent && _.isEqual(_.sortBy(Object.keys(value.eventData)), _.sortBy(eventKeys)) && _.isMatch(value.eventData, result)) {
+                                                return value;
+                                            }
+                                        })[0];
+                                        if (event) {
+                                            eventFound = true;
+                                            clearInterval(timer);
+                                            return resolve();
+                                        }
+                                    }
+                                }, timeoutConfigs.defaultInterval);
+                            });
+
+                            if (eventFound && event) {
+                                expect(event);
+                                expect(event.eventName);
+                                expect(event.eventName).to.equal(targetEvent);
+                                expect(event.eventData);
+
+                                for (const eventKey of eventKeys) {
+                                    expect(event.eventData).to.haveOwnProperty(eventKey);
+                                    if (eventKey === "projectID") {
+                                        expect(event.eventData[eventKey]).to.equal(projectID);
+                                    }
+                                    if (eventKey === "appStatus" || eventKey === "buildStatus") {
+                                        expect(event.eventData[eventKey]).to.equal(state);
+                                    }
+                                    if (eventKey === "appErrorStatus" || eventKey === "detailedBuildStatus") {
+                                        expect(event.eventData[eventKey]).to.equal(errMsg);
+                                    }
```

Comment:
We need to check the `detailedAppStatus` as well when we update the application status.

Reply:
The `detailedAppStatus` is not passed on the source code of the `updateStatus` function in line https://github.com/eclipse/codewind/blob/1681bf42aebda2594837632aa8db16fc91572d7a/src/pfe/file-watcher/server/src/controllers/projectStatusController.ts#L159

So we cannot test `detailedAppStatus` from `updateStatus`

Reply:
We call `updateProjectStatus` with `detailedAppStatus` when we ping app, but if we only test `updateStatus` that should be fine

Reply:
Yes, that is correct.